Javadoc target option 1.5 no longer supported under JDK 11+

```
$ mvn javadoc:javadoc
...
    [INFO] ------------------------------------------------------------------------
    [ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:2.7:jar (attach-javadocs) on project bytecode: MavenReportException: Error while creating archive:
    [ERROR] Exit code: 2 - error: Source option 5 is no longer supported. Use 6 or later.
    [ERROR] error: Target option 1.5 is no longer supported. Use 1.6 or later.
    [ERROR]
    [ERROR] Command line was: /usr/local/Cellar/openjdk@11/11.0.9/libexec/openjdk.jdk/Contents/Home/bin/javadoc @options @packages
```

--- a/pom.xml
+++ b/pom.xml
@@ -67,11 +67,11 @@
       <plugins>
 	<plugin>
 	  <artifactId>maven-clean-plugin</artifactId>
-	  <version>2.5</version>
+	  <version>3.1.0</version>
 	</plugin>
 	<plugin>
 	  <artifactId>maven-compiler-plugin</artifactId>
-	  <version>3.1</version>
+	  <version>3.8.1</version>
 	  <configuration>
 	    <source>${jdk.version}</source>
 	    <target>${jdk.version}</target>
@@ -79,31 +79,31 @@
 	</plugin>
 	<plugin>
 	  <artifactId>maven-dependency-plugin</artifactId>
-	  <version>2.7</version>
+	  <version>3.1.2</version>
 	</plugin>
 	<plugin>
 	  <artifactId>maven-deploy-plugin</artifactId>
-	  <version>2.8.1</version>
+	  <version>3.0.0-M1</version>
 	</plugin>
         <plugin>
           <artifactId>maven-gpg-plugin</artifactId>
-          <version>1.5</version>
+          <version>1.6</version>
         </plugin>
         <plugin>
           <artifactId>maven-install-plugin</artifactId>
-          <version>2.5.1</version>
+          <version>3.0.0-M1</version>
         </plugin>
 	<plugin>
 	  <artifactId>maven-jar-plugin</artifactId>
-	  <version>2.4</version>
+	  <version>3.2.0</version>
 	</plugin>
 	<plugin>
 	  <artifactId>maven-javadoc-plugin</artifactId>
-	  <version>3.0.1</version>
+	  <version>3.2.0</version>
 	  <configuration>
 	    <breakiterator>true</breakiterator>
 	    <quiet>true</quiet>
-	    <source>1.5</source>
+	    <source>${jdk.version}</source>
 	    <verbose>false</verbose>
 	    <linksource>true</linksource>
             <detectlinks>true</detectlinks>
@@ -114,22 +114,26 @@
 	</plugin>
 	<plugin>
 	  <artifactId>maven-release-plugin</artifactId>
-	  <version>2.5</version>
+	  <version>3.0.0-M1</version>
 	  <configuration>
 	    <autoVersionSubmodules>true</autoVersionSubmodules>
 	  </configuration>
 	</plugin>
         <plugin>
           <artifactId>maven-resources-plugin</artifactId>
-          <version>2.6</version>
+          <version>3.2.0</version>
         </plugin>
 	<plugin>
+	  <artifactId>maven-site-plugin</artifactId>
+	  <version>3.9.1</version>
+	</plugin>
+        <plugin>
 	  <artifactId>maven-source-plugin</artifactId>
-	  <version>2.2.1</version>
+	  <version>3.2.1</version>
 	</plugin>
 	<plugin>
 	  <artifactId>maven-surefire-plugin</artifactId>
-	  <version>3.0.0-M1</version>
+	  <version>3.0.0-M5</version>
           <configuration>
             <argLine>-Xmx1g</argLine>
           </configuration>
@@ -137,7 +141,7 @@
 	<plugin>
 	  <groupId>org.jvnet.jaxb2.maven2</groupId>
 	  <artifactId>maven-jaxb2-plugin</artifactId>
-	  <version>0.8.3</version>
+	  <version>0.14.0</version>
 	</plugin>
       </plugins>
     </pluginManagement>
@@ -181,12 +185,12 @@
     <plugins>
       <plugin>
 	<artifactId>maven-javadoc-plugin</artifactId>
-	<version>3.0.1</version>
+	<version>3.2.0</version>
 	<configuration>
 	  <aggregate>true</aggregate>
 	  <breakiterator>true</breakiterator>
 	  <quiet>true</quiet>
-	  <source>1.5</source>
+	  <source>${jdk.version}</source>
 	  <verbose>false</verbose>
 	  <linksource>true</linksource>
           <detectlinks>true</detectlinks>
@@ -197,7 +201,7 @@
       </plugin>
       <plugin>
 	<artifactId>maven-project-info-reports-plugin</artifactId>
-	<version>2.7</version>
+	<version>3.1.1</version>
         <reportSets>
           <reportSet>
             <reports>
@@ -225,4 +229,34 @@
     <module>sequencing</module>
     <module>phylo</module>
   </modules>
+  <profiles>
+    <profile>
+      <!-- Override maven plugin versions in release profile -->
+      <id>sonatype-oss-release</id>
+      <build>
+        <plugins>
+          <plugin>
+            <artifactId>maven-enforcer-plugin</artifactId>
+            <version>3.0.0-M5</version>
+          </plugin>
+          <plugin>
+            <artifactId>maven-gpg-plugin</artifactId>
+            <version>1.6</version>
+          </plugin>
+          <plugin>
+            <artifactId>maven-jar-plugin</artifactId>
+            <version>3.2.0</version>
+          </plugin>
+          <plugin>
+            <artifactId>maven-javadoc-plugin</artifactId>
+            <version>3.2.0</version>
+          </plugin>
+          <plugin>
+            <artifactId>maven-source-plugin</artifactId>
+            <version>3.2.1</version>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+  </profiles>
 </project>

